src/Taboo/TurnManager.py

```diff
@@ -91,29 +91,31 @@ def updateState(self, newState):
     # ---------------------------------
     # Message handlers
 
-    def processDiscard(self, qmsg):
-        """
-        Guarantees from the caller
-        1. qmsg.jmsg is of right length and right type
-        2. Correct player is invoking this
-        3. Game is not over yet
+    def __validateCompletedOrDiscard(self, qmsg):
+        """ Validates [COMPLETED|DISCARD, turn<int>, wordIdx<int>]
+        Replies a DISCARD-BAD or COMPLETED-BAD if the message is
+        received at wrong turn state
 
-        Always returns True (message is ingested)
+        Returns True iff message is valid
         """
+        msgType = qmsg.jmsg[0]
+        assert msgType in ("DISCARD", "COMPLETED")
+        badReplyType = "{}-BAD".format(msgType)
+
         ws = qmsg.initiatorWs
 
         if self._state != TurnMgrState.RUNNING:
-            trace(Level.play, "processDiscard turn state", self._state.name)
-            self._txQueue.put_nowait(ClientTxMsg(["DISCARD-BAD",
-                                                  "Can't discard right now"],
+            trace(Level.play, "process{} turn state".format(msgType), self._state.name)
+            self._txQueue.put_nowait(ClientTxMsg([badReplyType,
+                                                  "Can't {} right now".format(msgType)],
                                                  {ws}, initiatorWs=ws))
-            return True
+            return False
 
         if qmsg.jmsg[1] != self._curTurnId:
-            self._txQueue.put_nowait(ClientTxMsg(["DISCARD-BAD",
-                                                  "Discarding invalid turn"],
+            self._txQueue.put_nowait(ClientTxMsg([badReplyType,
+                                                  "invalid turn"],
```

Comment:
Error messages are sentence case everywhere. This used to be "Discarding invalid turn". Can we make this "Invalid turn" perhaps?

Reply:
Fixed in e39a7785eb889868aa5504df39efe2c5b681b651 and pushed.